lib/phoenix/socket/transport.ex

```diff
@@ -332,8 +332,13 @@ defmodule Phoenix.Socket.Transport do
     import Plug.Conn
     origin = conn |> get_req_header("origin") |> List.first()
     check_origin = check_origin_config(handler, endpoint, opts)
+    check_csrf = opts[:check_csrf]
 
     cond do
+      check_origin == false and check_csrf == false ->
```

Comment:
Is there any chance we can detect and raise this when the endpoint is being configured, rather than at request time?

Reply:
Technically we could do it at compile time somewhere like in https://github.com/phoenixframework/phoenix/blob/main/lib/phoenix/endpoint.ex#L637

However, `check_origin` is a runtime configuration option so there's not really to ensure it will no be changed at runtime. We could also check it when starting the endpoint, but as far as I can see we don't do this kind of checks in the current code.

This also raises the question of the `check_origin: {m, f, a}` configuration: should we raise if the MFA call returns `false` and `check_csrf` is set to `false` as well (not currently done)?

Reply:
Can we check it when the supervision starts or the transports start? if it is a MFA, it is fine to not check, then the user is really up their own devices after manually bypassing all layers.

Reply:
Done in d339626bd2e7be2198bda6bb34aeb7162453b227. Not sure it's the best place to do this, but I've pushed to discuss this :) There are no tests yet, hence I've set this PR to draft.